atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -181,6 +185,51 @@ public void dropTablesDeletesTableMetadata() throws TableMappingNotFoundExceptio
                         TEST_TABLE.getQualifiedName());
     }
 
+    @Test
+    public void dropTablesDeletesTableMetadataCaseInsensitivelyIfCaseInsensitiveSet()
+            throws TableMappingNotFoundException {
+        when(sqlConnection.selectCount(any(), any(), any())).thenReturn(1L);
+        createTable();
+        createOverflowTable();
+
+        tableMappingDdlTable.drop(false);
+
+        verify(sqlConnection)
+                .executeUnregisteredQuery(
+                        "DELETE FROM " + TABLE_MAPPING_DEFAULT_CONFIG.metadataTable() + " WHERE LOWER(table_name) ="
+                                + " LOWER(?)",
+                        TEST_TABLE.getQualifiedName());
+    }
+
+    @Test
+    public void dropTablesThrowsIfMultipleMatchingTableReferencesExistAndCaseInsensitiveDrop()
+            throws TableMappingNotFoundException {
+        long numberOfMatchingTableReferences = 2;
+        when(sqlConnection.selectCount(any(), any(), any())).thenReturn(numberOfMatchingTableReferences);
+        createTable();
+        createOverflowTable();
+
+        assertThatLoggableExceptionThrownBy(() -> tableMappingDdlTable.drop(false))
+                .hasLogMessage("There are multiple tables that have the same case insensitive table reference."
+                        + " Throwing to avoid accidentally deleting the wrong table reference."
+                        + " Please contact support to delete the metadata, which will involve deleting the row from"
+                        + " the DB manually.")
+                .hasExactlyArgs(
+                        SafeArg.of("numberOfMatchingTableReferences", numberOfMatchingTableReferences),
+                        UnsafeArg.of("tableReference", TEST_TABLE));
+    }
+
+    @Test
+    public void dropTablesDoesNotThrowIfMultipleMatchingTableReferencesExistAndCaseSensitiveDrop()
+            throws TableMappingNotFoundException {
+        // This mock is never called on the case sensitive code path, but this makes the test easier to read.
+        lenient().when(sqlConnection.selectCount(any(), any(), any())).thenReturn((long) 2);
+        createTable();
+        createOverflowTable();
+
+        assertThatCode(() -> tableMappingDdlTable.drop()).doesNotThrowAnyException();
```

Comment:
can we verify something about what was run? Not sure if this is also mocked away

Reply:
(this is already tested in `dropTablesDeletesTableMetadataCaseSensitively` and given that there aren't too many methods, we don't need to test it here)